pi4j-core/src/main/java/com/pi4j/context/impl/DefaultContext.java

```diff
@@ -133,7 +135,23 @@ public Context shutdown() throws ShutdownException {
         // shutdown the runtime
         this.runtime.shutdown();
         return this;
-    }
+	}
+
+	@Override
+	public <T extends IO> T shutdown(T io) {
+		if (null == io) {
+			return null;
+		}
+		return shutdown(io.getId());
+	}
+
+	@Override
+	public <T extends IO> T shutdown(String id) {
+		if (!hasIO(id)) {
```

Comment:
i would be defensive here, and throw an IllegalArgumentException

Reply:
I take that back. Looking at the implementation, exceptions are already thrown, remove the null checks and just call remove on the registry

Reply:
Removed the checks, added some Java doc and removed shutdown(IO io) because:

1. all other interfaces accepting just the string-id
2. avoid null pointer handling if the IO is null.